hbase-server/src/main/java/org/apache/hadoop/hbase/io/asyncfs/FanOutOneBlockAsyncDFSOutputSaslHelper.java

```diff
@@ -355,6 +353,55 @@ private void sendSaslMessage(ChannelHandlerContext ctx, byte[] payload) throws I
       return Collections.singletonList(new CipherOption(CipherSuite.AES_CTR_NOPADDING));
     }
 
+    private static class BuilderPayloadSetter {
+      private static Class<?> byteStringClass;
+      private static Class<?> builderClass;
+      private static Method copyFromMethod;
+      private static Method setPayloadMethod = null;
+
+      static void setter(DataTransferEncryptorMessageProto.Builder builder, byte[] payload) {
+        Object byteStringObject = null;
+        try {
+          byteStringObject = copyFromMethod.invoke(null, payload);
+        } catch (IllegalAccessException e) {
+          e.printStackTrace();
+        } catch (InvocationTargetException e) {
+          e.printStackTrace();
+        }
+
+        if (setPayloadMethod == null) {
+          try {
+            setPayloadMethod = builderClass.getMethod("setPayload", byteStringClass);
+          } catch (NoSuchMethodException e) {
+            e.printStackTrace();
+          }
+        }
+
+        try {
+          setPayloadMethod.invoke(builder, byteStringObject);
+        } catch (IllegalAccessException e) {
+          e.printStackTrace();
+        } catch (InvocationTargetException e) {
+          e.printStackTrace();
+        }
+      }
+
+      static {
+        builderClass = DataTransferEncryptorMessageProto.Builder.class;
+        byteStringClass = com.google.protobuf.ByteString.class;
```

Comment:
Is this ok? This presumes what pb is on the CLASSPATH? pb2.5?  Should it be the shaded hbase ByteString?

Reply:
Or this is just a default setting?

Reply:
Seems like it should be hbase internal shaded ByteString, not this com.google one.

Reply:
oh. i get it. This is the HDFS ByteString. The one that will be on the CP for all versions before HDFS3.3. Ok.